rca/editorial/models.py

```diff
@@ -0,0 +1,116 @@
+from wagtail.admin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    MultiFieldPanel,
+    PageChooserPanel,
+)
+from wagtail.api import APIField
+from wagtail.core.fields import RichTextField
+from wagtail.images.edit_handlers import ImageChooserPanel
+
+from rca.utils.models import BasePage, RelatedPage
+
+
+class EditorialPageRelatedSchoolsAndResearchPages(RelatedPage):
+    source_page = ParentalKey(
+        "EditorialPage", related_name="related_schools_and_research_pages"
+    )
+    panels = [
+        PageChooserPanel("page", ["schools.SchoolPage", "research.ResearchCentrePage"])
+    ]
+
+    api_fields = [APIField("page")]
+
+
+class Author(models.Model):
+    name = models.CharField(max_length=128)
+
+    def __str__(self):
+        return self.name
+
+
+class EditorialPageAuthor(models.Model):
+    page = ParentalKey("EditorialPage", related_name="editorial_authors")
+    author = models.ForeignKey("Author", related_name="+", on_delete=models.CASCADE)
+    panels = [FieldPanel("author")]
+
+    def __str__(self):
+        return self.author.name
+
+
+class EditorialPage(BasePage):
+
+    intro_text = RichTextField(blank=True, max_length=255)
+    #  this is not on the base page
+    hero_image = models.ForeignKey(
+        "images.CustomImage",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+    video = models.URLField(blank=True)
+    video_caption = models.CharField(
+        blank=True,
+        max_length=80,
+        help_text="The text displayed next to the video play button",
+    )
+    video_preview_image = models.ForeignKey(
+        "images.CustomImage",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    #  do you want it to be from date created - auto_now_add (I think )
+    #  or from date_updated = add_add
+    date = models.DateTimeField(auto_now_add=True)
+
+    # author - taxonomy - something to do with get_contenct on project page
+    # create author taxonomy
+    # foreign key to that taxoomy
+    #  look at how newstype is done inn wagtail kit
+    #  essnetially be doing the same thing for all texonomies
+    #  dont do the frontned stuff
+
+    #  create author app
+    #  create author model
+    #  do foreign key to that
+    #
+
+    # school_or_center = foreign key
+
+    email = models.EmailField(blank=True, max_length=254)
+
+    # def get_context(self, request, *args, **kwargs):
+    #     context = super().get_context(request, *args, **kwargs)
+    #     taxonomy_tags = []
+    #     if self.related_school_pages:
+    #         for i in self.related_school_pages.all():
+    #             taxonomy_tags.append({"title": i.page.title})
+    #     context["taxonomy_tags"] = taxonomy_tags
+
+    #     return context
+
+    # have not included date here as I tink it should be added automiatically
+    content_panels = BasePage.content_panels + [
```

Comment:
Missing 'Data; field from the content panels

Reply:
Do I want to add the date field? Will this not be added automatically from when it's published?

Reply:
If I add `auto_now_add` to the `datefield`, I think it makes the date non-editable and so it can't be added to the content panels. Unless, we don't use `auto_now_add` and the publisher can select a date for the Editorial Page?

Reply:
Hadn't realized auto_add_now would do that. Our AC states:

> As a publisher I can add a Date value to the Editorial page

So the main thing we need here is the ability to set an article date. We should also make this a required field too as it's what we will be automatically listing articles by. Maybe don't worry about auto_add_now. Strictly speaking, it's not a requirement but a nice to have